src/test/java/segelzwerg/sporttooolbox/SpeedCalculatorTest.java

```diff
@@ -20,4 +20,14 @@ public void thirty_kilometer_one_hour_test_speed() {
 
         assertThat(speed, equalTo(expectedSpeed));
     }
+
+    @Test
+    public void add_distance_test() {
+        Distance thirtyKilometer = new Distance(30,999);
+        Distance thirtyonekilometer= thirtyKilometer.addDistance(new Distance(0,1));
+
+        Distance expected=new Distance(31,0);
+
+        assertThat(thirtyonekilometer, equalTo(expected));
+    }
 }
```

Comment:
This should be in DistanceTest so we later can find test easier corresponding to the classes.
And you are almost right. The distance class was missing an annotation for creating the equals and hashCode method. This was my bad, but I fixed in 4a714a3

Reply:
I saw a different issue raised for this so am not touching this part.